Assets/Scripts/Map/PatrollingMap.cs

```diff
@@ -15,7 +15,7 @@ public sealed class PatrollingMap : ICloneable<PatrollingMap>
     {
         public readonly IReadOnlyList<Vertex> Vertices;
 
-        public readonly Dictionary<int, Vertex[]> VerticesByPartition;
+        public readonly List<Partition> Partitions;
```

Comment:
It should be a readonlylist you should not be able to add / remove elements from the list after patrollingmap has been generated.

Reply:
makes sense, changed it.